app/models/host.rb

```diff
@@ -30,7 +30,11 @@ class Host < ApplicationRecord
   validates :account, presence: true
 
   def all_profiles
-    (assigned_profiles + test_result_profiles).uniq
+    [assigned_profiles, test_result_profiles].flat_map do |profiles|
+      scoped = profiles
+      scoped = yield scoped if block_given?
+      scoped.nil? ? profiles : scoped
+    end.uniq
```

Comment:
Could we simplify this a bit? I think it'd be great to have this return an `ActiveRecord::Relation` so that we can use it like this: `Host.first.all_profiles.in_policy(policy_or_profile_id)`. I find the block here pretty confusing; it doesn't yield a single profile but the same collection passed into the flat_map (i.e. `all_profiles`). It's admittedly an idiom, but it's usually used in creating new objects, like `Config.new do |config| ...`, rather than working with collections of objects.

Reply:
For it to return `ActiveRecord::Relation` it needs to do a UNION. UNION is not builtin in Rails AFAIK. Even if it was, the joins done by `in_policy` (was `policy_bunch`) scope would need to be applied for each UNION sub query. Which leads to the logic of two yields. I'd be happy to have it return a Relation. Could you suggest changes?

It yields twice because there are two collections/relations. I find it straightforward and effective. Do you have some improvement in mind, considering that the method cannot return single Relation?

Reply:
Doesn't this work?

```suggestion
    Profile.where(id: assigned_profiles).or(Profile.where(id: test_result_profiles))
```

Reply:
That ... might ... actually ... work! Thanks!

Reply:
Changed.